### PR TITLE
Map GL status ONCHAIN to models::ChannelState::Closed

### PR DESCRIPTION
**Pull request by ok300**
Now two different GL states map a channel state to `models::ChannelState::Closed`: `ONCHAIN` and `CLOSED`.

### CODE REVIEW

File: libs/sdk-core/src/greenlight/node_api.rs [hunk 1]
```diff
@@ -929,16 +923,16 @@ fn parse_amount(amount_str: String) -> Result<pb::Amount> {
...
+impl From<pb::Channel> for Channel {
     fn from(c: pb::Channel) -> Self {
         let state = match c.state.as_str() {
-            "OPENINGD" | "CHANNELD_AWAITING_LOCKIN" => crate::models::ChannelState::PendingOpen,
-            "CHANNELD_NORMAL" => crate::models::ChannelState::Opened,
-            "CLOSED" => crate::models::ChannelState::Closed,
-            _ => crate::models::ChannelState::PendingClose,
+            "OPENINGD" | "CHANNELD_AWAITING_LOCKIN" => ChannelState::PendingOpen,
+            "CHANNELD_NORMAL" => ChannelState::Opened,
+            "ONCHAIN" | "CLOSED" => ChannelState::Closed,
```
Comment (roeierez) [inline libs/sdk-core/src/greenlight/node_api.rs:931]: According to core lightning docs there is no status CLOSED https://docs.corelightning.org/reference/lightning-listpeers

**Review by roeierez — CHANGES_REQUESTED** [review]
Looks good. Just need to remove the CLOSED status as far as I understand the docs

File: libs/sdk-core/src/greenlight/node_api.rs (hunk 1 above)
Comment (ok300) [reply libs/sdk-core/src/greenlight/node_api.rs:931]: If I search that link for `"CLOSED"` I find:

> The objects in the channels array will have at least these fields:
> - state: Any of these strings: 
>   ...
>   -  "CLOSED": The channel closure has been confirmed deeply. The channel will eventually be removed from this array.

Is this not the same status we're interested in?

Comment (JssDWt) [reply libs/sdk-core/src/greenlight/node_api.rs:931]: @ok300 I agree

**Review by JssDWt — APPROVED** [review]

File: libs/sdk-core/src/greenlight/node_api.rs (hunk 1 above)
Comment (roeierez) [reply libs/sdk-core/src/greenlight/node_api.rs:931]: Right .  My miss.

**Review by roeierez — APPROVED** [review]

**Pull request merged**